tensorflow_datasets/core/dataset_builder.py

```diff
@@ -72,6 +72,8 @@ class BuilderConfig(object):
   `BuilderConfig` and add their own properties.
   """
 
+  __slots__ = ["_name", "_version", "_supported_versions", "_description"]
```

Comment:
For `BuilderConfig`, this seems over-optimisation. How much does it actually save ? Unless you have a dataset with millions of configs, I don't really understand where it would change something.

Additionally, this class is frequently subclassed.

TensorInfo & GraphRunner are good though.

Reply:
Good point, I removed it in e3f7366